smart_open/tests/test_smart_open.py

```diff
@@ -354,6 +354,64 @@ def test_read_str_from_bytes(self):
             data = sf.read()
         self.assertEqual(data, SAMPLE_TEXT)
 
+    def test_read_str_from_bytes_api_rt(self):
+        """Can we read strings from a byte stream?"""
+        buffer = make_buffer(initial_value=SAMPLE_BYTES)
+        with smart_open.smart_open(buffer, 'rt') as sf:
+            data = sf.read()
+        self.assertEqual(data, SAMPLE_TEXT)
+
+    def test_read_str_from_bytes_api_a(self):
+        """Can we read strings from a byte stream?"""
+        buffer = make_buffer()
+        with smart_open.smart_open("test", 'wb') as sf:
```

Comment:
Please use the tempfile module to generate temporary file names. Do this in the setUp method.

Writing to a local file `test` all the time is a bad idea because it increases the potential of coupling between tests.